extensions/wikia/Email/Controller/DiscussionController.class.php

```diff
@@ -53,6 +53,27 @@ public function body() {
         ] );
     }
 
+    public function assertCanEmail() {
+        parent::assertCanEmail();
+        $this->assertSubscribedToDiscussionsEmail();
+    }
+
+    /**
+     * Asserts that target user is subscribed to Discussions emails.
+     *
+     * @throws \Email\Check
+     */
+    protected function assertSubscribedToDiscussionsEmail() {
+        $wantsDiscussionEmails = $this->targetUser->getLocalPreference(
```

Comment:
We're doing a check against a local preference here, but down below it looks like we're only operating on the global preference. Should this be a global check instead? If so, we should update `$wgDefaultUserOptions["enotifdiscussions-$cityId"] = 1;` as well.